### PR TITLE
Refactoring FF Router

### LINKED ISSUE #564: traffic_constraints ignored
When testing the schedule function I wanted to "temp_close" all doors with by traffic_constraints. But this does not work, the traffic_constraints were ignored. Using an event file to "temp_close" the doors works. But it's a bug anyway.

 [Archiv.zip](https://github.com/JuPedSim/jpscore/files/3879130/Archiv.zip)

### PR DESCRIPTION
**Pull request by schroedtert**
- remove wrong error message from router, when ped is already outside
- remove not needed code
- adding missing documentation
- fix clang-tidy warnings (some)
- at start up compute all floor fields even if door is closed, penalise closed doors with increased edge weight

Addresses some of the issue mentioned in #374.

Fixes #564.

### CODE REVIEW

File: libcore/src/Simulation.cpp [hunk 1]
```diff
@@ -313,19 +325,21 @@ void Simulation::UpdateRoutesAndLocations()
...
-        int subRoomID  = ped->GetSubRoomID();
-
-        if(auto cross = dynamic_cast<Crossing *>(target)) {
-            if(cross->IsInRoom(roomID) && cross->IsInSubRoom(subRoomID)) {
-                if(!ped->IsWaiting() && cross->IsTempClose()) {
-                    ped->StartWaiting();
-                }
+        int goal = ped->FindRoute();
+        if(goal != FINAL_DEST_OUT) {
+            Hline * target = _building->GetTransOrCrossByUID(goal);
```
Comment (zeroset) [inline libcore/src/Simulation.cpp]: const?

File: libcore/src/Simulation.cpp [hunk 2]
```diff
@@ -313,19 +325,21 @@ void Simulation::UpdateRoutesAndLocations()
...
+            int roomID     = ped->GetRoomID();
+            int subRoomID  = ped->GetSubRoomID();
+
+            if(auto cross = dynamic_cast<Crossing *>(target)) {
```
Comment (zeroset) [inline libcore/src/Simulation.cpp]: const

File: libcore/src/Simulation.cpp (hunk 1 above)
Comment (schroedtert) [reply libcore/src/Simulation.cpp]: done

File: libcore/src/Simulation.cpp (hunk 2 above)
Comment (schroedtert) [reply libcore/src/Simulation.cpp]: done

**Pull request merged**